src/utils/work/fetchWork.js

```diff
@@ -87,22 +94,29 @@ const fetchGeneExpressionWork = async (
 const fetchWork = async (
   experimentId,
   body,
-  backendStatus,
+  getState,
   optionals = {},
 ) => {
   const { extras = undefined, timeout = 180, eventCallback = null } = optionals;
+  const backendStatus = getState().backendStatus[experimentId].status;
+  const { environment } = getState().networkResources;
 
   if (!isBrowser) {
     throw new Error('Disabling network interaction on server');
   }
 
   const { pipeline: { startDate: qcPipelineStartDate } } = backendStatus;
   if (body.name === 'GeneExpression') {
-    return fetchGeneExpressionWork(experimentId, timeout, body, backendStatus, extras);
+    return fetchGeneExpressionWork(experimentId, timeout, body, backendStatus, environment, extras);
+  }
+
+  let cacheUniquenessKey = null;
+  if (environment !== Environment.PRODUCTION && localStorage.getItem('disableCache') === true) {
+    cacheUniquenessKey = Math.random();
```

Comment:
I would also put a log message here, printing that we are going to randomise the cachekey, because caching is disabled